### PR TITLE
Bugzilla-77129: display received time in email list

### PR DESCRIPTION
**Pull request by k-kato**
[Adding feature]
- If zimbraPrefDisplayTimeInMailList is TRUE, web client displays not only date but also time of a message received yesterday or before in email list.
- Adding a setting in Preferences -> Mail tab

[Related]
- Jira ticket id: NICPS-30
- Pull requests:
-- https://github.com/Zimbra/zm-web-client/pull/227
-- https://github.com/Zimbra/zm-mailbox/pull/528

### CODE REVIEW

File: WebRoot/js/ajax/util/AjxDateUtil.js [hunk 1]
```diff
@@ -283,21 +283,22 @@ function(date) {
...
+	var time = AjxDateUtil.computeTimeString(date);
 	if (now.getTime() - dateMSec < AjxDateUtil.MSEC_PER_DAY &&
 		now.getDay() == date.getDay()) {
-		return AjxDateUtil.computeTimeString(date);
+		return time;
 	}
 
 	if (now.getFullYear() == date.getFullYear()) {
-		return AjxDateUtil._dateFormatNoYear.format(date);
+		return AjxDateUtil._dateFormatNoYear.format(date) + (requireTime ? " " + time : "");
```
Comment (silentsakky) [inline WebRoot/js/ajax/util/AjxDateUtil.js]: I am not sure if there could be some languages which wants to jumble up sequence of date and time parts, if that is the case this logic will fail

Comment (silentsakky) [reply WebRoot/js/ajax/util/AjxDateUtil.js]: Can we use different formatters in case when we want to show time also?

Comment (k-kato) [reply WebRoot/js/ajax/util/AjxDateUtil.js]: @silentsakky I will change it to use another (new?) formatter.

**Comment by k-kato** [comment]
two key-value pairs are added in the following format:
- AjxMsg.formatDateTimeMediumNoYear
=>"AjxMsg.formatDateMediumNoYear + [whitespace] + I18nMsg.formatTimeShort"
- I18nMsg.formatDateTimeShort
=> "I18nMsg.formatDateShort + [whitespace] + I18nMsg.formatTimeShort"

When zimbraPrefDisplayTimeInMailList is FALSE, time is not shown in the email list.

(format)
- received today: I18nMsg.formatTimeShort
- received this year: AjxMsg.formatDateMediumNoYear
- received last year or before: I18nMsg.formatDateShort

When zimbraPrefDisplayTimeInMailList is TRUE, time is shown in the email list.

(format)
- received today: I18nMsg.formatTimeShort
- received this year: AjxMsg.formatDateTimeMediumNoYear
- received last year or before: I18nMsg.formatDateTimeShort

The format may be wrong in some language/locale. Even so, we will be able to fix it by editing a properties file.

**Review by silentsakky — APPROVED** [review]

**Pull request closed without merging**